### PR TITLE
fix(query): negative rate/increase due to NaN chunk

### PR DESCRIPTION
**Pull request by sherali42**
**Pull Request checklist**

- [x] The commit(s) message(s) follows the contribution [guidelines](CONTRIBUTING.md) ?
- [ ] Tests for the changes have been added (for bug fixes / features) ?

This change is to handle a case where a time series ends with a NaN value. Currently this results into a new chunk/row with a single NaN value. 
As we default NaN to zero value during rate/increase window calculation, this sets highestValue to be zero. This is resulting into negative rate/increase values, since rate = highestValue-lowestValue.
This fix is to short `addTimeChunks` operation and return without any effect on highestValue calculation.

### CODE REVIEW

File: query/src/main/scala/filodb/query/exec/rangefn/RateFunctions.scala [hunk 1]
```diff
@@ -173,6 +173,10 @@ abstract class ChunkedRateFunctionBase extends CounterChunkedRangeFunction[Trans
                     startRowNum: Int, endRowNum: Int,
                     startTime: Long, endTime: Long): Unit = {
     val dblReader = reader.asDoubleReader
+    // if the chunk is a single row NaN value, then return. Prometheus end of time series marker.
+    // This is to make sure we don't set highestValue to zero. avoids negative rate/increase values.
+    if (startRowNum == 0 && endRowNum == 0 && dblReader.apply(acc, vector, startRowNum).isNaN)
```
Comment (yu-shipit) [inline query/src/main/scala/filodb/query/exec/rangefn/RateFunctions.scala:178]: Just curious if check should be
 `if (startRowNum == endRowNum && dblReader.apply(acc, vector, startRowNum).isNaN)`

Comment (sherali42) [reply query/src/main/scala/filodb/query/exec/rangefn/RateFunctions.scala:178]: for this issue to manifest, it has to be single row NaN chunk. We could just test for `endRowNum == 0` this implies startRowNum is 0.

Comment (sandeep6189) [reply query/src/main/scala/filodb/query/exec/rangefn/RateFunctions.scala:178]: Thanks @sherali42 . So basically, we are not updating the highestValue in this condition ? 
Should we also add a unit test for this edge case ?

**Review by sandeep6189 — APPROVED** [review]

File: query/src/main/scala/filodb/query/exec/rangefn/RateFunctions.scala (hunk 1 above)
Comment (sherali42) [reply query/src/main/scala/filodb/query/exec/rangefn/RateFunctions.scala:178]: @sandeep6189 added a test case, PTAL

**Review by sandeep6189 — APPROVED** [review]
Thank you @sherali42 for taking this up 👍
